src/setup-beam.js

```diff
@@ -278,11 +266,24 @@ async function getRebar3Versions() {
 }
 
 function getVersionFromSpec(spec, versions) {
-  if (versions.includes(spec)) {
-    return spec
+  if (core.getInput('version-type', { required: false }) === 'strict') {
+    if (versions.includes(spec)) {
+      return spec
+    }
+    return null
   }
 
-  return semver.maxSatisfying(versions, semver.validRange(spec))
+  // We keep a map of semver => actualver in order to use semver ranges to find appropriate versions
```

Comment:
... and the new way we keep references in a map for further handling. Tests should be the stuff giving us confidence over this, so we should improve/add to those if you feel it's warranted.